src/vs/workbench/contrib/searchEditor/browser/searchEditor.ts

```diff
@@ -248,6 +248,15 @@ export class SearchEditor extends BaseTextEditor {
 		}
 	}
 
+	focusSearchInput() {
+		const viewState = this.loadViewState();
+		if (viewState && viewState.focused === 'editor') {
+			this.queryEditorWidget.searchInput.focus();
+		} else {
+			this.searchResultEditor.focus();
```

Comment:
No real need to make this toggle, it can simply call this.queryEditorWidget.searchInput.focus();

Reply:
https://github.com/microsoft/vscode/pull/94799/commits/23aeb705457f1da59513272732f7df6223fccf92